packages/eslint-plugin/src/rules/member-ordering.ts

```diff
@@ -615,6 +615,39 @@ function getRank(
   return getRankOrder(memberGroups, orderConfig);
 }
 
+function changeToGroups(
```

Comment:
[Style] I hard a really hard time reading through this function and how it's used. Which is not really this PR's fault; `member-ordering` is an annoyingly complex rule by definition. But to make it easier for future us, I think this function would be more understandable with:

* A name like `groupMembers` or `groupMembersBy(something)`, where that _something_ is what they're actually being grouped on... `Type`?
* A JSDoc comment explaining what it does

WDYT?